### DISCUSSION TITLE
ros2 humble(fastdds v2.6.9) can use dds-router?

### DISCUSSION THREAD

**Discussion by Autostone-c** [eProsima/DDS-Router#486] [Q&A]
My project is based on Ros2 Humble. I now want to use DDS router for isolation and WAN communication, but I encountered many errors during the source code compilation, which seems to be an issue with the FastDDS version. If I want to compile DDS router on FastDDS (v2.6.9), which version(dds-router) should I use?

**Comment by juanlofer-eprosima** [comment]
Hi @Autostone-c, indeed if trying to compile a recent version of DDS-Router a higher Fast-DDS version is required. Your options are:
- Build DDS-Router [v0.3.0](https://github.com/eProsima/DDS-Router/releases/tag/v0.3.0) against the Fast-DDS present in ROS. [These](https://github.com/eProsima/DDS-Router/blob/v0.3.0/ddsrouter.repos) are the dependencies for that version, the one for Fast-DDS is not exactly the same but should probably work.
- Build a recent 2.x DDS-Router [(v2.2.0](https://github.com/eProsima/DDS-Router/releases/tag/v2.2.0) for example, avoid 3.x due to prevent incompatibilites), adding to your workspace the required Fast-DDS version (do not use the one present in ROS).
- Use instead a [Vulcanexus](https://vulcanexus.org/) humble [docker image](https://hub.docker.com/r/eprosima/vulcanexus/), which is our ROS 2 distribution with the latest middleware features and bugfixes, additionally including useful tools for developing and monitoring ROS2 applications. Cloud and desktop tags will contain DDS-Router application.
